tests/runtime/filter_aws.c

```diff
@@ -576,17 +552,249 @@ void flb_test_aws_ec2_tags_value_500() {
     flb_stop(ctx);
     flb_aws_client_mock_destroy_generator();
     flb_destroy(ctx);
-    flb_sds_destroy(name);
     flb_free(ops);
 
     set_output(NULL);
 }
 
+void flb_test_aws_ec2_tags_include() {
+    int ret;
+    int bytes;
+    char *p = "[0, {\"log\": \"hello, from my ec2 instance\"}]";
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    int filter_ffd;
+    struct flb_lib_out_cb cb_data;
+    struct flb_aws_client_generator *client_generator;
+    struct flb_filter_aws_init_options *ops;
+    struct flb_aws_client_mock_request_chain *request_chain;
+    char *output = NULL;
+    char *result;
+
+    request_chain = FLB_AWS_CLIENT_MOCK(
+        response(
+            expect(URI, "/latest/meta-data/tags/instance"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "Name\nCUSTOMER_ID"),
+            set(PAYLOAD_SIZE, 16)
+        ),
+        response(
+            expect(URI, "/latest/meta-data/tags/instance/Name"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "my_ec2_instance"),
+            set(PAYLOAD_SIZE, 15)
+        ),
+        response(
+            expect(URI, "/latest/meta-data/tags/instance/CUSTOMER_ID"),
+            expect(METHOD, FLB_HTTP_GET),
+            set(STATUS, 200),
+            set(PAYLOAD, "70ec5c04-3a6e-11ed-a261-0242ac120002"),
+            set(PAYLOAD_SIZE, 36)
+        )
+    );
+    flb_aws_client_mock_configure_generator(request_chain);
+
+    client_generator = flb_aws_client_get_mock_generator();
+    ops = flb_calloc(1, sizeof(struct flb_filter_aws_init_options));
```

Comment:
check alloc failure?

Reply:
I know this is a test and the other tests don't have checks but its best to add them... even if the alloc failure wouldn't be likely to happen and we could just retry the test, its good ot have all the code following best practices in open source since others might follow or copy it..

Reply:
Sure, I added a check for the NULL case.
```c
    ops = flb_calloc(1, sizeof(struct flb_filter_aws_init_options));
    if (ops == NULL) {
        TEST_MSG("calloc for aws plugin options failed\n");
        TEST_CHECK(false);
        return;
    }
```